assets/js/frontend/course-video/youtube-extension.js

```diff
@@ -0,0 +1,79 @@
+( () => {
+	function handleVisibilityChange( player ) {
+		return function () {
+			if ( document.hidden ) {
+				player.pauseVideo();
+			}
+		};
+	}
+
+	function preventClick( event ) {
+		event.preventDefault();
+		return false;
+	}
+
+	function onYouTubePlayerStateChange( event ) {
+		const playerStatus = event.data;
+
+		if (
+			window.sensei.courseVideoSettings.courseVideoRequired &&
```

Comment:
Not critical too, but I wonder if it would be interesting to destructure these variables from `courseVideoSettings` in the scope of the anonymous function, just to make the code a little cleaner.

Reply:
Fixed: https://github.com/Automattic/sensei/pull/4546/commits/18d7b4e8d4884972749e2b1acbee71c2a471d70b